Objects/moduleobject.c

```diff
@@ -841,11 +841,22 @@ _Py_module_getattro_impl(PyModuleObject *m, PyObject *name, int suppress)
         }
         if (suppress != 1) {
             if (_PyModuleSpec_IsInitializing(spec)) {
-                PyErr_Format(PyExc_AttributeError,
+                origin = PyObject_GetAttr(spec, &_Py_ID(origin));
```

Comment:
`PyObject_GetAttr()` can set not only AttributeError. If it sets a different error, it should not be overridden by AttributeError.

I suggest to use `PyObject_GetOptionalAttr()`.

Reply:
Why not chain the original error to a new exception (raise from)?